### PR TITLE
Fix NO_FILE_BASED_LOGGING environment variable behavior

### LINKED ISSUE #9161: File-Based Logging During Upgrade Ignores Kolibri Setting
## Observed behavior

Despite the `KOLIBRI_NO_FILE_BASED_LOGGING` envar being set, Kolibri tries to write to logs during upgrade.

## Errors and logs

```
Traceback (most recent call last):
  File "/usr/local/bin/kolibri", line 5, in <module>
    from kolibri.utils.cli import main
  File "/usr/local/lib/python3.6/site-packages/kolibri/utils/cli.py", line 17, in <module>
    from kolibri.plugins import config
  File "/usr/local/lib/python3.6/site-packages/kolibri/plugins/__init__.py", line 16, in <module>
    from kolibri.utils.conf import KOLIBRI_HOME
  File "/usr/local/lib/python3.6/site-packages/kolibri/utils/conf.py", line 47, in <module>
    os.mkdir(LOG_ROOT)
OSError: [Errno 30] Read-only file system: '/home/kolibri/.kolibri/logs'
```

## Expected behavior

Would assume that it Kolibri does not try to log to file when the `NO_FILE_BASED_LOGGING` setting is set to `True`

## Steps to reproduce

1. Set the `KOLIBRI_NO_FILE_BASED_LOGGING` envar to true
2. Run any command that triggers upgrade. `kolibri manage migrate` in my case

## Context

Was discovered during CD, in a Linux container of Ubuntu LTS 18.

### PR DESCRIPTION
**Pull request by KshitijThareja**
## Summary
This PR aims to fix the behavior of `NO_FILE_BASED_LOGGING` environment variable.

## References
Closes #9161 

## Reviewer guidance
To test these changes:

1. Set the `KOLIBRI_NO_FILE_BASED_LOGGING` env variable to true.
2. Run any command that triggers upgrade, for example- `kolibri manage migrate`.
3. You'll see that _"kolibri.txt"_ file (or _"debug.txt"_ file) in the _"logs"_ folder doesn't reflect the update logs generated.
4. If the _"logs"_ folder doesn't already exist and the env variable is set to true, there will be no attempt made to create it automatically.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by KshitijThareja** [comment]
@rtibbles I've fixed the part where error was being thrown when the environament variable was set to true and it was still trying to create a _logs_ directory when the `KOLIBRI_HOME `path was read-only. But even after this, if the path is read-only, there will be an error related to the creation of _logs_ folder and also errors related to database access. 
I can add a condition to make sure that if the path is read only, we don't attempt to create a new logs folder. But I don't think we can prevent the database errors(as it obvious we need write-access for it to work properly).
What are your views on this?

**Comment by rtibbles** [comment]
The `NO_FILE_BASED_LOGGING` is the fix to handle a read only environment, along with either setting a specific writable path for a SQLite DB, or using PostgreSQL as a backend, so I don't think we need to handle that!

File: kolibri/utils/conf.py
Comment (rtibbles) [inline kolibri/utils/conf.py]: No need to do the extra check here, if `NO_FILE_BASED_LOGGING` is set to any value, that's enough. So can just simplify this to:
```
if not NO_FILE_BASED_LOGGING:
```

(or even combine with the if statement above for the path exist check).

File: kolibri/utils/logger.py
Comment (rtibbles) [inline kolibri/utils/logger.py]: No need to update this.

File: kolibri/utils/logger.py
Comment (rtibbles) [inline kolibri/utils/logger.py:208]: Does defining these handlers even if unused cause any errors? Would be nice to minimize the diff here if not needed.

**Review by rtibbles — CHANGES_REQUESTED** [review]
No need to update how we check the env var - one question about whether the diff is needed in one instance.

File: kolibri/utils/conf.py
Comment (KshitijThareja) [reply kolibri/utils/conf.py]: Sure. I just added the extra condition just in case if the user sets the environment variable to false explicitly. In that case, without this condition, the "logs" folder would still be created. But if we don't need to consider that, I'll remove the extra conditionals.

File: kolibri/utils/logger.py
Comment (rtibbles) [reply kolibri/utils/logger.py:208]: Just want to verify the answer to this question here - does defining the handlers, even if they are unused cause any errors?

Comment (rtibbles) [reply kolibri/utils/logger.py:208]: Bumping this question ^

**Comment by MisRob** [comment]
Hi @KshitijThareja, I see there's unresolved conversation but you also asked for a new review - could you elaborate a bit on where things are and if you're/aren't waiting for us?

**Comment by KshitijThareja** [comment]
Hey @MisRob.
The changes are complete from my end. I am just waiting for the final review. Also, @rtibbles had enquired about the use of handlers this way (in my changes), which in my opinion is necessary as otherwise it would result in errors, for the reasons already mentioned in my previous comments.
If there are any more changes required in this PR, I'd be happy to work on them. If not, I guess we can consider it as completed.

**Comment by rtibbles** [comment]
Hi @KshitijThareja - it wasn't clear to me based on your previous responses that just setting up (and not using) the handlers specifically was problematic. I have tested locally to confirm this, but I asked the question to seek this confirmation.

**Review by rtibbles — APPROVED** [review]

**Comment by KshitijThareja** [comment]
Hi @rtibbles. I'm sorry for any misunderstanding from my end. 
> just setting up (and not using) the handlers specifically was problematic.

Regarding this, yes, doing so would be problematic. That's what I had intended to convey initially. 
Also, thanks for approving the changes :)  
If there's anymore input required from my side, I'd be happy to do so  😄

**Pull request merged**
